pdf/core/encoding.go

```diff
@@ -1335,6 +1481,110 @@ func (this *RawEncoder) EncodeBytes(data []byte) ([]byte, error) {
 }
 
 //
+// CCITTFax encoder/decoder (dummy, for now)
+//
+type CCITTFaxEncoder struct{}
+
+func NewCCITTFaxEncoder() *CCITTFaxEncoder {
+	return &CCITTFaxEncoder{}
+}
+
+func (this *CCITTFaxEncoder) GetFilterName() string {
+	return StreamEncodingFilterNameCCITTFax
+}
+
+func (this *CCITTFaxEncoder) MakeDecodeParams() PdfObject {
+	return nil
+}
+
+// Make a new instance of an encoding dictionary for a stream object.
+func (this *CCITTFaxEncoder) MakeStreamDict() *PdfObjectDictionary {
+	return MakeDict()
+}
+
+func (this *CCITTFaxEncoder) DecodeBytes(encoded []byte) ([]byte, error) {
+	return encoded, nil
+}
+
+func (this *CCITTFaxEncoder) DecodeStream(streamObj *PdfObjectStream) ([]byte, error) {
+	return streamObj.Stream, nil
+}
+
+func (this *CCITTFaxEncoder) EncodeBytes(data []byte) ([]byte, error) {
+	return data, nil
+}
+
+//
+// JBIG2 encoder/decoder (dummy, for now)
+//
+type JBIG2Encoder struct{}
+
+var ErrNoJBIG2 = errors.New("JBIG2Encoder is not implemented")
+
+func NewJBIG2Encoder() *JBIG2Encoder {
+	return &JBIG2Encoder{}
+}
+
+func (this *JBIG2Encoder) GetFilterName() string {
+	return StreamEncodingFilterNameJBIG2
+}
+
+func (this *JBIG2Encoder) MakeDecodeParams() PdfObject {
+	return nil
+}
+
+// Make a new instance of an encoding dictionary for a stream object.
+func (this *JBIG2Encoder) MakeStreamDict() *PdfObjectDictionary {
+	return MakeDict()
+}
+
+func (this *JBIG2Encoder) DecodeBytes(encoded []byte) ([]byte, error) {
+	return encoded, nil
+}
+
+func (this *JBIG2Encoder) DecodeStream(streamObj *PdfObjectStream) ([]byte, error) {
+	return streamObj.Stream, nil
+}
+
+func (this *JBIG2Encoder) EncodeBytes(data []byte) ([]byte, error) {
+	return data, nil
+}
+
+//
+// JPX encoder/decoder (dummy, for now)
+//
+type JPXEncoder struct{}
```

Comment:
Define an ErrNoJPXEncoder,  or simply define an ErrUnsupportedEncoding, which can be used for all Dummy encodings until they are implemented.

Also, return those errors when attempting to use dummy implementations.